sql/json-functions-generated-column.md

```diff
@@ -0,0 +1,117 @@
+```
+
+`JSON_EXTRACT` is a function that TiDB supports. If you are familiar with JSON in MySQL 5.7, you must know it. The `JSON_EXTRACT` function in TiDB behaves the same as in MySQL. The meaning of the above function is to extract the `city` field from the `address_info` document. The second argument is a "path expression" and is used to specify which field to extract. The complete syntax description of the path expression is complex. Here are a few simple examples to help you understand its general use:
+
+```sql
+SET @person = '{"name":"John","friends":[{"name":"Forest","age":16},{"name":"Zhang San","gender":"male"}]}';
+
+SELECT JSON_EXTRACT(@person,  '$.name'); -- gets "John"
+SELECT JSON_EXTRACT(@person,  '$.friends[0].age'); -- gets 16
+SELECT JSON_EXTRACT(@person,  '$.friends[1].gender'); -- gets "male"
+SELECT JSON_EXTRACT(@person,  '$.friends[2].name'); -- gets NULL
+``` 
+
+In addition to inserting and querying data, TiDB also support editing JSON. In general, TiDB currently supports the following JSON functions in MySQL 5.7:
+
+- [JSON_EXTRACT](https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-extract)
+- [JSON_ARRAY](https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-array)
+- [JSON_OBJECT](https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-object)
+- [JSON_SET](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-set)
+- [JSON_REPLACE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-replace)
+- [JSON_INSERT](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-insert)
+- [JSON_REMOVE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-remove)
+- [JSON_TYPE](https://dev.mysql.com/doc/refman/5.7/en/json-attribute-functions.html#function_json-type)
+- [JSON_UNQUOTE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-unquote)
+
+You can get the general use of these functions directly from the function name. These functions in TiDB behave the same as in MySQL 5.7. If you want to query the specific use, see the [JSON Functions document of MySQL 5.7](https://dev.mysql.com/doc/refman/5.7/en/json-functions.html). If you are a user of MySQL 5.7, you can migrate seamlessly.
+
+Currently TiDB does not support all the JSON functions in MySQL 5.7. This is because our preliminary goal is to provide complete support for **MySQL X Plugin**, which covers the majority of JSON functions used to insert, select, update and delete data. More functions will be supported if necessary.
+
+## Index JSON Using Generated Column
+
+The full table scan is executed when you query a JSON field. When you run `EXPLAIN` statements in TiDB, the results show that it is full table scan. Then, can you index the JSON field?
```

Comment:
When you run `EXPLAIN` statements in TiDB ->
When you run the `EXPLAIN` statement in TiDB